tests/common/graceful_shutdown_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
```

Comment:
Thanks for using common framework, it's pretty nice use case, however I'm not sure if proposed approach of putting `e2e` tests with `//go:build e2e` into `common` test directory. I would leave it in `tests/e2e` for now.

Reply:
I wanted to verify this test assertion with integration test as well. But I encountered the problem that etcd server created by integration test stopped abruptly without leadership transfer. After fixing this, integration test timed out. More details in https://github.com/etcd-io/etcd/pull/15828#issuecomment-1535521820. 

This build tag will be eventually removed when the test timed out issue is resolved.

Reply:
I didn't propose to remove the `//go:build e2e` but to move the test to `./tests/e2e` instead.

Reply:
Got it. Thanks!